### PR TITLE
3

### PR DESCRIPTION
**Pull request by XOctOpus1**

### CODE REVIEW

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Please push your code. Because for now only one space is changed

File: app/actions/preparations.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    if character["potion"] is not None:
+        if "power" in character["potion"]["effect"]:
+            character["power"] += character["potion"]["effect"]["power"]
+
+        if "protection" in character["potion"]["effect"]:
+            character["protection"] \
+                += character["potion"]["effect"]["protection"]
+
+        if "hp" in character["potion"]["effect"]:
+            character["hp"] += character["potion"]["effect"]["hp"]
```
Comment (ArsenPidhoretskyi) [inline app/actions/preparations.py:20]: Use loop instead of duplicate code

File: app/main.py [hunk 2]
```diff
@@ -1,206 +1,33 @@
...
-    # check if someone fell in battle
-    if arthur["hp"] <= 0:
-        arthur["hp"] = 0
-
-    if red_knight["hp"] <= 0:
-        red_knight["hp"] = 0
+    fight(arthur, red_knight)
 
     # Return battle results:
     return {
```
Comment (ArsenPidhoretskyi) [inline app/main.py:33]: You can use for loop

File: app/main.py [hunk 3]
```diff
@@ -1,206 +1,33 @@
...
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+    red_knight = knightsconfig["red_knight"]
+    prepare(red_knight)
```
Comment (ArsenPidhoretskyi) [inline app/main.py:21]: You can create a loop with knights and use prepare function for each

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by Grayder0152 — CHANGES_REQUESTED** [review]

**Review by Grayder0152 — APPROVED** [review]

**Pull request closed without merging**